src/main.c

```diff
@@ -0,0 +1,147 @@
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/shell/shell.h>
+#include <stdlib.h>
+
+LOG_MODULE_REGISTER(main);
+
+// Define stepper motor parameters
+#define STEPS 500
+#define STEP_DELAY 10 // Step delay in milliseconds
+#define MAX_DATA_POINTS 100 // Maximum data points to collect
+
+#define ZEPHYR_USER_NODE DT_PATH(zephyr_user)
+
+const struct gpio_dt_spec step = GPIO_DT_SPEC_GET(ZEPHYR_USER_NODE, step_gpios);
+const struct gpio_dt_spec dir = GPIO_DT_SPEC_GET(ZEPHYR_USER_NODE, dir_gpios);
+
+const struct device *sensor = DEVICE_DT_GET_ANY(ti_tmag5273);
+
+// Structure to store sensor data
+struct sensor_data {
+    int x;
+    int y;
+    int z;
+};
+
+// Array to store sensor data for each step
+struct sensor_data sensor_values[MAX_DATA_POINTS];
+int data_count = 0; // Counter to keep track of stored steps
+
+// Function to store sensor data for up to 100 steps
+void store_sensor_data(void)
+{
+    if (data_count >= MAX_DATA_POINTS) {
+        LOG_INF("Data collection complete, reached maximum steps of %d", MAX_DATA_POINTS);
+        return;
+    }
+
+    struct sensor_value value_x, value_y, value_z;
+
+    if (sensor_sample_fetch(sensor) == 0 &&
+        sensor_channel_get(sensor, SENSOR_CHAN_MAGN_X, &value_x) == 0 &&
+        sensor_channel_get(sensor, SENSOR_CHAN_MAGN_Y, &value_y) == 0 &&
+        sensor_channel_get(sensor, SENSOR_CHAN_MAGN_Z, &value_z) == 0) {
+
+        sensor_values[data_count].x = value_x.val1;
```

Comment:
`value_x.val1` is only part of the raw data. To get a correct value with a unit, there is a calculation required. This calculation is already implemented in the driver, but you have to trigger it.

Try to convert the values like this:
```
double magn_y = sensor_value_to_double(&value);
LOG_INF("Magnetometer Y-axis: %lf", magn_y);
```

Your structure `sensor_data` has to hold double instead of int after the change. As a resulting value, you should get values with the unit G (Gauss), which you could then convert to Tesla, I guess that unit is more common.